ff/serving/metadata/metadata.go

```diff
@@ -112,6 +114,38 @@ func (id ResourceID) Parent() (ResourceID, bool) {
 	}, true
 }
 
+var bannedStrings = [...]string{"__"}
+var bannedPrefixes = [...]string{"_"}
+var bannedSuffixes = [...]string{"_"}
+
+func resourceNamedSafely(id ResourceID) error {
+	for _, substr := range bannedStrings {
+		if len(id.Name) >= len(substr) && strings.Contains(id.Name, substr) {
+			return fmt.Errorf("resource name %s contains banned string %s", id.Name, substr)
+		}
+		if len(id.Variant) >= len(substr) && strings.Contains(id.Variant, substr) {
+			return fmt.Errorf("resource variant %s contains banned string %s", id.Name, substr)
+		}
+	}
+	for _, substr := range bannedPrefixes {
```

Comment:
I believe there a strings.suffix and strings.prefix function

Reply:
good call